subsys/bluetooth/shell/bt.c

```diff
@@ -1217,6 +1219,284 @@ static int cmd_adv_info(const struct shell *shell, size_t argc, char *argv[])
+		return -EINVAL;
+	}
+
+	if (!strcmp(argv[1], "off")) {
+		if (bt_le_per_adv_stop(adv) < 0) {
+			shell_error(shell,
+				    "Failed to stop periodic advertising");
+		} else {
+			shell_print(shell, "Periodic advertising stopped");
+		}
+	} else if (!strcmp(argv[1], "on")) {
+		if (bt_le_per_adv_start(adv) < 0) {
+			shell_error(shell,
+				    "Failed to start periodic advertising");
+		} else {
+			shell_print(shell, "Periodic advertising started");
+		}
+	} else {
+		shell_error(shell, "Invalid argument: %s", argv[1]);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int cmd_per_adv_param(const struct shell *shell, size_t argc,
+			     char *argv[])
+{
+	struct bt_le_ext_adv *adv = adv_sets[selected_adv];
+	struct bt_le_per_adv_param param;
+	int err;
+
+	if (!adv) {
+		shell_error(shell, "No extended advertisement set selected");
+		return -EINVAL;
+	}
+
+	if (argc > 1) {
+		param.interval_min = strtol(argv[1], NULL, 16);
+	} else {
+		param.interval_min = BT_GAP_ADV_SLOW_INT_MIN;
+	}
+
+	if (argc > 2) {
+		param.interval_max = strtol(argv[2], NULL, 16);
+	} else {
+		param.interval_max = param.interval_min * 1.2;
+
+	}
+
+	if (param.interval_min > param.interval_max) {
+		shell_error(shell,
+			    "Min interval shall be less than max interval");
+		return -EINVAL;
+	}
+
+	if (argc > 3 && !strcmp(argv[3], "tx-power")) {
+		param.options = BT_LE_ADV_OPT_USE_TX_POWER;
+	} else {
+		param.options = 0;
+	}
+
+	err = bt_le_per_adv_set_param(adv, &param);
+	if (err) {
+		shell_error(shell, "Failed to set periodic advertising "
+			    "parameters (%d)", err);
+		return -ENOEXEC;
+	}
+
+	return 0;
+}
+
+static int cmd_per_adv_data(const struct shell *shell, size_t argc,
+			    char *argv[])
+{
+	struct bt_le_ext_adv *adv = adv_sets[selected_adv];
+	static uint8_t ad[256];
+	uint8_t ad_len = 0;
+	int err;
+
+	if (!adv) {
+		shell_error(shell, "No extended advertisement set selected");
+		return -EINVAL;
+	}
+
+	ad_len = hex2bin(argv[1], strlen(argv[1]), ad, sizeof(ad));
+
+	if (!ad_len) {
+		shell_error(shell, "Could not parse adv data");
+		return -ENOEXEC;
+	}
+
+	err = bt_le_per_adv_set_data(adv, ad, ad_len);
+	if (err) {
+		shell_error(shell,
+			    "Failed to set periodic advertising data (%d)",
+			    err);
+		return -ENOEXEC;
+	}
+
+	return 0;
+}
+#endif /* CONFIG_BT_PER_ADV */
+
+#if defined(CONFIG_BT_PER_ADV_SYNC)
+static struct bt_le_per_adv_sync *per_adv_syncs[CONFIG_BT_PER_ADV_SYNC_MAX];
+static struct bt_le_per_adv_sync *selected_per_adv_sync;
+
+static void per_adv_sync_sync_cb(struct bt_le_per_adv_sync *sync,
+				 struct bt_le_per_adv_sync_synced_info *info)
+{
+	char le_addr[BT_ADDR_LE_STR_LEN];
+
+	if (selected_per_adv_sync != sync) {
+		shell_error(ctx_shell, "Invalid per adv sync object");
+		return;
+	}
+
+	bt_addr_le_to_str(info->addr, le_addr, sizeof(le_addr));
+	shell_print(ctx_shell, "PER_ADV_SYNC[%u]: [DEVICE]: %s synced, "
+		    "Interval %u, PHY %u",
```

Comment:
Interval is printed as hex other places, I would suggest to do the same here to avoid confusion, could also print it in actual ms value as well, e.g:
```
		shell_print(ctx_shell, "Interval: 0x%04x (%u ms)",
			    info.le.interval, info.le.interval * 5 / 4);
```

Reply:
Fixed